### ISSUE TITLE
IE raises "Access denied" error in some cases

### ISSUE THREAD

**Issue by gaearon** [rodneyrehm/viewport-units-buggyfill#38]
I'm getting occasional localized errors in my JS logs translating as "Access denied" from IE 11 with stacks like this:

```js
Error: Toegang geweigerd. // Error: Access denied

  at findProperties (app/~/viewport-units-buggyfill/viewport-units-buggyfill.js?:189)
  at func (app/~/viewport-units-buggyfill/viewport-units-buggyfill.js?:164)
  at func (app/~/viewport-units-buggyfill/viewport-units-buggyfill.js?:90)
```

I don't think the line numbers are very accurate, but from my understand this is caused by [accessing `document.styleSheets` when some stylesheet is loaded from another host](https://lists.w3.org/Archives/Public/www-dom/2003AprJun/0007.html).

Shall we try-catch it?

**Comment by rodneyrehm** [comment]
we already have something in place to deal with foreign resources: [importCrossOriginLinks](https://github.com/rodneyrehm/viewport-units-buggyfill/blob/master/viewport-units-buggyfill.js#L341-L363). We should try making that work before simply silencing something. Maybe the try-catch then becomes unnecessary?

**Comment by gaearon** [comment]
The problem is `document.styleSheets` itself throws in some cases (maybe has to do with IE permissions). How do we work around that?

**Comment by rodneyrehm** [comment]
it probably throws when accessing something you shouldn't. the only way I see this would work is by dopping the `forEach` and going with a `for () { try{ … } catch(e){} }` loop. do you want to try it out and send a PR?

**Comment by gaearon** [comment]
Submitted as #39.

I have not tested this yet (written in browser).
I also don't like complete error swallowing, shall we at least do `console.warn`?

**Comment by jabes** [comment]
I have made this PR which fixed this issue for me. I received this error in IE9 after calling sheet.cssRules on a remote stylesheet from inside an iframe. Since the CSSRuleList should always be an array-like object, it is better to use typeof.

**Comment by jabes** [comment]
https://github.com/rodneyrehm/viewport-units-buggyfill/pull/64

**Comment by rodneyrehm** [comment]
fixed by merging #64
